### ISSUE TITLE
k6 gets stuck when executed in ConEmu

### ISSUE THREAD

**Issue by pamela-ruiz** [grafana/k6#2056]
I used the test script in the documentation, I run with k6 run script.js and it gets stuck.
I have to externally cancel the process because ctrl + C doesn't stop it.
It looks like this:
![image](https://user-images.githubusercontent.com/56687554/121400894-7d222380-c92e-11eb-929f-b530997cf3db.png)

version of k6: 0.32.0
OS: Windows 10

**Comment by na--** [comment]
Which test script? :confused:

**Comment by pamela-ruiz** [comment]
This one:

import http from 'k6/http';
import { sleep } from 'k6';

export default function () {
  http.get('http://test.k6.io');
  sleep(1);
}

I downgraded to the previous version, and it worked

**Comment by na--** [comment]
This is very strange :confused: Can you please run the script with `k6 run --verbose script.js` and paste the resulting log messages from running it with k6 v0.32.0 here? I can't replicate this issue, but I'm running on Linux.

Also, can you adjust the script slightly:
```js
import http from 'k6/http';
import { sleep } from 'k6';

export default function () {
    console.log('start');
    http.get('http://test.k6.io');
    console.log('middle');
    sleep(1);
    console.log('end');
}
```
just so we understand where this script is getting stuck on... :confused:

**Comment by tom-miseur** [comment]
I've just tested this script on Windows 10 with v0.32 and it worked as expected.

**Comment by pamela-ruiz** [comment]
Yes, k6 gets stuck and never finishes, even if I wait the default timeout or press Ctrl + C.
I've installed k6 with the .msi package from the official page https://dl.k6.io/msi/k6-latest-amd64.msi.

**Comment by na--** [comment]
I have no idea what could be happening and why we can't reproduce this... :confused: Any help or ideas will be much appreciated! :pray:

**Comment by tom-miseur** [comment]
@pamela-ruiz could you try copying and running the script from another location, say `C:\Temp`?

**Comment by imiric** [comment]
@pamela-ruiz Which terminal/shell are you using? The `#` prompt in your screenshot is rare to see on Windows.

**Comment by MStoykov** [comment]
Hi, I would like you to provide us with the stacktrace after it freezes ... but unfortunately, it seems this isn't an easy on windows. On non-windows you can have hit `ctrl+\` to send SIGQUIT which will dump them and stop the process .. but not on windows :( .

I have build an exe(in this [zip](https://github.com/k6io/k6/files/6629735/k6-v0.32.0-1-g5f05fb9c-windows-amd64.zip)) from https://github.com/k6io/k6/commit/5f05fb9c32a5e2fe3750230c94d29e33fe36ce84 which v0.32.0 + the change to dump the stacktrace on `ctrl+c`, hopefully, it will capture that. It will be a lot of text and we care ... probably about most of it so please copy all of it, sorry for the trouble :(.

Can you also try other terminals if possible and tell us what you tried and how it worked in hopes we can narrow it down.

**Comment by pamela-ruiz** [comment]
Hi! I was using ConEmu, when I tried with normal windows cmd it worked. So maybe there is a problem with ConEmu.

**Comment by imiric** [comment]
@pamela-ruiz I'm not able to reproduce this with ConEmu 210422, the script runs fine for me:

![2021-06-10-144403_1387x647_scrot](https://user-images.githubusercontent.com/1009277/121527527-c9d33080-c9fa-11eb-8c12-7e6a2763944a.png)

You also seem to be running an admin instance of cmd.exe judging from the `#` prompt. In general this isn't a good idea, but it shouldn't be relevant in this case.

Can you please try the .exe @MStoykov uploaded above, press Ctrl+C when it gets stuck and send us the _entire_ output? ConEmu has an option under Edit>Copy all that will allow you to save it to a text file and upload it here.

**Comment by pamela-ruiz** [comment]
[output.txt](https://github.com/k6io/k6/files/6631614/output.txt)

Sure, here is the output

**Comment by MStoykov** [comment]
@pamela-ruiz, this looks ... really confusing ... it looks like it's just ... having trouble doing the network calls for HTTP :thinking: 

 Can you please wait a minute or two(more is better :) ) before killing it so hopefully it has more relevant data and just to make certain all ... most timeouts has fired.

**Comment by pamela-ruiz** [comment]
Sure, no problem here is the new output.
[output2.txt](https://github.com/k6io/k6/files/6631796/output2.txt)

**Comment by pamela-ruiz** [comment]
Awesome, that worked.
`k6 -q --no-color run script.js`

**Comment by na--** [comment]
Lowering the priority of this, since it seems like such a specific issue

**Comment by testn** [comment]
Sounds like https://conemu.github.io/en/MicrosoftBugs.html#ctrl-c

**Comment by ptiseo-tegria** [comment]
First-time K6 user. I am experiencing this same issue on ConEMU 210912 [x64] Stable. Win10 20H2 (a.ka. 19042.1766), K6 v0.39 with my first script, using the smoke test (modified, of course) on the site as a sample. K6 simply hangs with no changes even if I let it go for many minutes. Using `--no-color` alone still leads to a hang. Using `-q` alone seems to allow it to complete. 

Script:
```
import http from "k6/http";
import { check, sleep } from "k6";

export const options = {
  vus: 1, // 1 user looping for 1 minute
  duration: "1m",

  thresholds: {
    http_req_duration: ["p(99)<1500"], // 99% of requests must complete below 1.5s
  },
};

const TST_API_BASE_URL = "https://something";
const TST_ACCESS_TOKEN = "sometoken";

export default () => {
  const authHeaders = {
    headers: {
      Authorization: `Bearer ${TST_ACCESS_TOKEN}`,
      "Content-Type": "application/json",
      Accept: "application/json",
    },
  };

  const resp = http.get(`${TST_API_BASE_URL}/status`, authHeaders);
  check(resp, {
    "is 200 OK": (r) => r.status === 200,
  });

  sleep(1);
};
```

**Comment by na--** [comment]
> Using `--no-color` alone still leads to a hang. Using `-q` alone seems to allow it to complete.

This is super weird... :confused: I guess it might be the UTF characters like ✓ that cause ConEmu to hang? Or linux newlines? :thinking: Anyone has any ideas on how to solve it?

**Comment by na--** [comment]
@ptiseo-tegria (and others suffering from the same issue), please consider reporting this to ConEmu, since it's most likely a problem with it, not with k6: https://github.com/Maximus5/ConEmu/issues It may even be already reported, though there are tons of open issues, so I am not sure... :disappointed: 

In any case, since it's almost certain that there is nothing we can do on the k6 side to resolve this issue by default, I'll close it. We have another issue to discuss a new option that modifies how the k6 CLI UI is rendered, and I added a note about it there: https://github.com/grafana/k6/issues/1376#issuecomment-1184122967

**Comment by jeme** [comment]
New to K6 as well, I ran into this very issue and I wanted to add some additional info. This is just with the perhaps most simple example of:

```
import http from 'k6/http';
import { sleep } from 'k6';

export default function () {
  http.get('https://localhost:7088/');
  sleep(1);
}
```

Downgrading to K6 v.20 worked in ConEmu, in the default console v.40 works fine.
Further, If I in ConEmu run a Bash (Msys2-64) tab, v.40 works fine in ConEmu as well.
Did not test furter.

**Comment by mstoykov** [comment]
@jeme is v0.20.0 the newest version that does work for you?

From everything this looks like a conEmu issue not a k6 one. 

So please open an issue with them and if you can pinpoint the version that broke it we likely can at least try to guess which change is triggering this. 

Although again some people (on our team) have tried to reproduce this, and it has worked for them, so the problem seems to be a lot more complicated than the title fo the issue suggests.

**Comment by jeme** [comment]
@mstoykov Not necessarily - I ended up just going far back and taking a "random one" which then worked, I agree that this is likely a ConEmu issue.

HOWEVER I wanted to add the additional information, **specifically** about the Bash workaround for others who might come across this issue so they have more ways forward. I am perfectly happy with the workarounds!

One thought was if this could be made easier to find, as I did spend quite a bit of time finding this issue which meant I was frustrated and stuck for a while, now I can be rather stubborn at times so I stuck with it but I could imagine others giving up sooner in that process which would be a shape for this project :S... But I don't really have any good ideas on how to make this kind of information easier to find.
